### ISSUE TITLE
More clearly define "user facing bugs"

### ISSUE THREAD

**Issue by benjimelito** [ga-wdi-boston/game-project#720]
I believe we should explicitly state that NOT displaying an error message to the user when something goes wrong is a user facing bug. This seems to be something that trips up a lot of people, and I feel that it's something that could be avoided if we let them know we are looking for it.

**Comment by BenGitsCode** [comment]
I think I agree, but we don't hold them to that standard as closely in Project 1 as we do in 2. Should we?

I wonder if this would be better as an addition the project repo or as a small talk about user-facing bugs? I like them learning those nuances as they go better for some reason, but I can't justify as to why.

**Comment by MicFin** [comment]
Right now, I believe we have not been giving any DNM if the only bugs are no messages for failed sign up, sign in, change password, etc.  But, if they have any other bugs that make it a DNM then we add the missing messaging as a requirement for re-submittal.

There has been discussion about having more "user story" type requirements for Project 1 where potentially it would end up looking something like:
`As a user, I want to sign up so I can play tic tac toe`
`As a user, I want to sign up with an email, password and password confirmation so I can sign in`
`As a user, I want to sign up and receive a message if the email is already taken or the password and password confirmation do not match so I know to try sign up again`

This could potentially help with the problem.  Still in early discussions though.

**Comment by gaand** [comment]
Any action that provides no feedback on success or failure is a UFB.  Feedback includes hiding/showing things.

**Comment by MicFin** [comment]
In that case, I think the suggestion to be more explicit about what a user facing bug is a good suggestion because they have little experience or context to know that a lack of messaging is a user facing bug.

@BenGitsCode said:
>I wonder if this would be better as an addition the project repo or as a small talk about user-facing bugs

We can not cover or identify all potential user facing bugs that they could create, but we can give them a few examples that we know they are likely to encounter like success and failure handling.  Potentially we can add a link after:
>Have any user-facing bugs. (link here)

The link could go to a section in the [tips.md](https://github.com/ga-wdi-boston/game-project/blob/master/tips.md) that covers "Common User Facing Bugs" where we can cover some basics like buttons that don't do anything, forms that don't clear, failing ajax response messaging, etc.

**Comment by BenGitsCode** [comment]
Yeah I didn't mean anything more than `Here are some of the most common UFBs in WDI, how to avoid them, and why they're bad.`

I think putting those in that `tips.md` there would be a great place for them.

**Comment by payne-chris-r** [comment]
Could also go in the `Developer Ten Commandments`

**Comment by jrhorn424** [comment]
I don't want to define UFB. I also want to the ask them if they've ever used a web application before when they ask what a UFB is.

I'm ok keeping a manifest of common bugs, so long as it's explained that no such list can ever be exhaustive.

Also in that tips file, they should be instructed on how to test for UFBs. The only sure-fire way I know for **them** to do so is to do things out-of-order, i.e. test the "sad path".
